producers/yarn_audit/types/yarn-issue.go

```diff
@@ -174,51 +152,86 @@ type contact struct {
 	Name string `json: name`
 }
 
-// YarnAuditReport includes yarn audit data grouped by advisories, actions and summary
-type YarnAuditReport struct {
-	AuditAdvisories []*auditAdvisoryData
-	AuditActions  []*auditActionData
-	AuditSummary       *auditSummaryData
-}
+func NewReport(report []byte) (*[]AuditAction, *[]AuditAdvisory, *[]AuditSummary, error) {
 
-// NewReport returns a YarnAuditReport, assuming each line is jsonline and returns any errors
-func NewReport(reportLines [][]byte) (*YarnAuditReport, []error) {
+	var raws []json.RawMessage
+	if err := json.Unmarshal(report, &raws); err != nil {
+		return nil, nil, nil, err
+	}
 
-	var report YarnAuditReport
+	var auditActions []AuditAction
+	var auditAdvisories []AuditAdvisory
+	var auditSummaries []AuditSummary
 
-	var errors []error
+	for _, raw := range raws {
+		auditAction := new(AuditAction)
+		if auditAction.Unmarshal(raw) {
+			auditActions = append(auditActions, *auditAction)
+		}
```

Comment:
I don't like that we're suppressing json unmarsheling errors here. For example if the structure of fields is modified by yarn audit, we wouldn't know.